paasta_tools/cli/cmds/validate.py

```diff
@@ -601,91 +720,80 @@ def validate_autoscaling_configs(service_path):
                 )
                 is False
             ):
-                autoscaling_params = instance_config.get_autoscaling_params()
-                if autoscaling_params["metrics_provider"] == "active-requests":
-                    desired_active_requests_per_replica = autoscaling_params.get(
-                        "desired_active_requests_per_replica",
-                        DEFAULT_DESIRED_ACTIVE_REQUESTS_PER_REPLICA,
-                    )
-                    if desired_active_requests_per_replica <= 0:
-                        returncode = False
-                        print(
-                            failure(
-                                msg="Autoscaling configuration is invalid: desired_active_requests_per_replica must be "
-                                "greater than zero",
-                                link="",
-                            )
-                        )
-                    if len(instance_config.get_registrations()) > 1:
-                        returncode = False
-                        print(
-                            failure(
-                                msg="Autoscaling configuration is invalid: active-requests autoscaler doesn't support instances with multiple registrations.",
-                                link="",
-                            )
-                        )
-                if autoscaling_params["metrics_provider"] in {
-                    "uwsgi",
-                    "piscina",
-                }:
-                    # a service may omit both of these keys, but we provide our own
-                    # default setpoint for all metrics providers so we are safe to
-                    # unconditionally read it
-                    setpoint = autoscaling_params["setpoint"]
-                    if setpoint <= 0:
-                        returncode = False
-                        print(
-                            failure(
-                                msg="Autoscaling configuration is invalid: setpoint must be greater than zero",
-                                link="",
-                            )
-                        )
+                autoscaling_params = instance_config.get_autoscaling_params()  # type: ignore
```

Comment:
it's too bad that mypy doesn't see the cast above and apply the cast here as well :p

Reply:
Oh I can fix that.